packages/react/src/search/Search.tsx

```diff
@@ -0,0 +1,58 @@
+import { useComposedRefs } from "@radix-ui/react-compose-refs";
+import { useControllableState } from "@radix-ui/react-use-controllable-state";
+import { type ComponentPropsWithRef, forwardRef, useRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Button } from "../button";
+import { Input } from "../input";
+import { IconCross } from "./IconCross";
+import { IconSearch } from "./IconSearch";
+
+type SearchProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  Omit<ComponentPropsWithRef<typeof Input>, "onChange">,
+  {
+    defaultValue?: string;
+    onChange?: (value: string) => void;
+    value?: string;
+  }
+>;
+
+export const Search = forwardRef<HTMLInputElement, SearchProps>(
+  ({ defaultValue = "", onChange, value: valueProp, ...props }, outerRef) => {
+    const innerRef = useRef<HTMLInputElement>(null);
+    const ref = useComposedRefs(innerRef, outerRef);
+
+    const [value, setValue] = useControllableState({
+      defaultProp: defaultValue,
+      onChange: onChange,
+      prop: valueProp,
+    });
```

Comment:
i don't think this is the right approach - your previous approach was the right one

so there's two ways to go about this - either you expose completely custom `onValueChange` and `onValueClear` props (or call them `onChange` and `onClear` in which case you prevent allowing outsiders to pass in their own `onChange` to the actual `<input />`)

or you keep the existing `onChange` handler and when clearing you simply call it again after manually clearing the value on the input

and as for controlled/uncontrolled - you can just do that based on the `typeof valueProp`
```suggestion
    const [innerValue, setValue] = useState(defaultValue);
    const value = typeof valueProp === "undefined" ? innerValue : valueProp;
```